test/python/http_recorder.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""Utilities (based on VCRpy) to record remote requets and allow testing offline/cached."""
+
+import json
+from contextlib import suppress
+from vcr.persisters.filesystem import FilesystemPersister
+from vcr import VCR
+
+
+class IdRemoverPersister(FilesystemPersister):
+    """
+    IdRemoverPersister is a VCR persister. This is, it implements a way to save and load cassettes.
+    This persister in particular inherits load_cassette from FilesystemPersister (basically, it
+    loads a standard cassette in the standard way from the file system). On the saving side, it
+    replaces some fields in the JSON content of the responses with dummy values.
+    """
+
+    @staticmethod
+    def get_responses_with(string_to_find, cassette_dict):
+        """
+        Filters the requests from cassette_dict
+
+        Args:
+            string_to_find (str): request path
+            cassette_dict (dict): a VCR cassette dictionary
+
+        Returns:
+            Request: VCR's representation of a request.
+        """
+        request_indices = [i for i, request in enumerate(cassette_dict['requests']) if
+                           string_to_find in request.path]
+        return [cassette_dict['responses'][i] for i in request_indices]
+
+    @staticmethod
+    def get_new_id(field, path, id_tracker, type_=str):
+        """
+        Creates a new dummy id (or value) for replacing an existing id (or value).
+
+        Args:
+            field (str): field name is used, in same cases, to create a dummy value.
+            path (str): path of the request is used, in same cases, to create a dummy value.
+            id_tracker (dict): a map of already assigned ids and generated ids.
+            type_ (type): type of the value.
+
+        Returns:
+            str: that is used to replace a value.
+        """
+
+        if type_ == float:
+            return 0.42
+        if type_ == int:
+            return 42
+        dummy_name = 'dummy%s%s' % (path.replace('/', ''), field)
+        count = len(list(filter(lambda x: str(x).startswith(dummy_name), id_tracker.values())))
+        return "%s%02d" % (dummy_name, count + 1)
+
+    @staticmethod
+    def get_maching_dicts(data_dict, map_list):
+        """
+        Find subdicts that are described in map_list.
+        Args:
+            data_dict (dict): in which the map_list is going to be searched.
+            map_list (list): the list of nested keys to find in the data_dict
+
+        Returns:
+            list: a list of dictionaries, each of them matches map_list.
+        """
+        ret = []
+        if not map_list:
+            return ret
+        if isinstance(data_dict, list):
+            _ = [ret.extend(IdRemoverPersister.get_maching_dicts(i, map_list)) for i in data_dict]
```

Comment:
Convert this into a loop, please. It is easier to read and name `i` in a meaningful way.

Reply:
9eebae8